transform/variant/tests/test_maf_transform.py

```diff
@@ -0,0 +1,192 @@
+                             allowable_codes=(200, 400, 404),
+                             allowable_methods=('GET', 'POST'))
+
+
+@pytest.fixture
+def maf_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test.maf')
+
+
+@pytest.fixture
+def gz_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'gz-test.maf.gz')
+
+
+@pytest.fixture
+def no_center_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-no-center.maf')
+
+
+@pytest.fixture
+def LARP6_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-LARP6.maf')
+
+
+@pytest.fixture
+def TMX3_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-TMX3.maf')
+
+
+@pytest.fixture
+def TPTE_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-TPTE.maf')
+
+
+@pytest.fixture
+def AXIN1_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-AXIN1.maf')
+
+
+@pytest.fixture
+def RIMS1_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-RIMS1.maf')
+
+
+@pytest.fixture
+def ZFP3_file(request):
+    """ get the full path of the test fixture """
+    return os.path.join(request.fspath.dirname, 'test-ZFP3.maf')
+
+
+
+@pytest.fixture
+def emitter_path_prefix(request):
+    """ get the full path of the test output """
+    return os.path.join(request.fspath.dirname, 'test')
+
+
+def validate(maf_file, emitter_path_prefix):
+    allele_file = '{}.Allele.Vertex.json'.format(emitter_path_prefix)
+    allelecall_file = '{}.AlleleCall.Edge.json'.format(emitter_path_prefix)
+    callset_file = '{}.Callset.Vertex.json'.format(emitter_path_prefix)
+    # remove output
+    with contextlib.suppress(FileNotFoundError):
+        os.remove(allele_file)
+        os.remove(allelecall_file)
+        os.remove(callset_file)
+    # create output
+    maf_transform.convert(maf_file, emitter_path_prefix)
+    # test.Allele.Vertex.json
+    error_message = 'maf_transform.convert({}, {}) should create {}' \
+                    .format(maf_file, emitter_path_prefix, allele_file)
+    assert os.path.isfile(allele_file),  error_message
+    # test allele contents
+    with open(allele_file, 'r', encoding='utf-8') as f:
+        for line in f:
+            # should be json
+            allele = json.loads(line)
+            # minimum protograph keys
+            assert list(allele.keys()) == ['gid', 'label', 'data'], \
+                'expected keys'
+            # should not be emoty
+            for k in allele.keys():
+                assert allele[k], 'empty key %s' % k
```

Comment:
The checks above are great. Maybe this is something we should define a pattern for. Does ratify have all these checks already?

Reply:
Yes ratify has similar 'should exist' type tests.  However, ratify uses protograph config as basis for 'graph'